llvm/include/llvm/ADT/APFloat.h

```diff
@@ -1536,6 +1536,26 @@ inline APFloat abs(APFloat X) {
   return X;
 }
 
+/// Returns X^N for N >= 0.
+inline APFloat pow(const APFloat &X, int64_t N) {
```

Comment:
`pow` is absolutely the *wrong* name for this function, since there is a standard `pow` function that this does not implement. `powi` (which is the LLVM intrinsic name) or `pown` would work better, though I worry that this is too numerically inaccurate to be a correct implementation of `pown`.

Reply:
Thanks @jcranmer-intel . I will rename to `powi`.

Reply:
> `pown` would work better, though I worry that this is too numerically inaccurate to be a correct implementation of `pown`

@jcranmer-intel do we have some reference implementation of `pown` in llvm?

Reply:
I don't think we do: https://libc.llvm.org/headers/math/index.html

Reply:
I don't think we should add this unless it's correctly rounded